File: 525. Contiguous Array/_525.py
class Solution:
    def findMaxLength(self, nums) -> int:
        maxLength = 0
        myDict = dict()
        count = 0
        myDict[0] = 0

        for i,v in enumerate(nums, 1):
            if v == 0:
                count -= 1
            else:
                count += 1

            tmp = myDict.get(count, -1)
            if tmp == -1:
                myDict[count] = i
            else:
                maxLength = max(maxLength, i-tmp)

        return maxLength

File: 525. Contiguous Array/525.go
package leetcode

func findMaxLength(nums []int) int {
    myDict := make(map[int]int, 0)
	count := 0
	maxLength := 0
	myDict[0] = 0

	for i,v := range nums{
		if v == 0{
			count --
		}else{
			count ++
		}

		i ++
		
		if tmp, ok := myDict[count];!ok{
			myDict[count] = i
		}else{
			if i-tmp > maxLength{
				maxLength = i-tmp
			}
		}
	}

	return maxLength
}